.mergify.yml

```diff
@@ -2,16 +2,27 @@ queue_rules:
   - name: default
     conditions: &ci_checks
       # Conditions to get out of the queue (= merged)
-      - status-success="Run test suite (centos7, pip, openmpi)"
-      - status-success="Run test suite (centos7, pip, mpich)"
-      - status-success="Run test suite (rockylinux8, pip, openmpi)"
-      - status-success="Run test suite (rockylinux8, pip, mpich)"
-      - status-success="Run test suite (rockylinux8, conda, openmpi)"
-      - status-success="Run test suite (rockylinux8, conda, mpich)"
-      - status-success="Run test suite (ubuntu2004, pip, openmpi)"
-      - status-success="Run test suite (ubuntu2004, pip, mpich)"
-      - status-success="Run test suite (ubuntu2004, conda, openmpi)"
-      - status-success="Run test suite (ubuntu2004, conda, mpich)"
+      - status-success="Run test suite (centos7, pip, openmpi, 3.7)"
+      - status-success="Run test suite (centos7, pip, openmpi, 3.8)"
+      - status-success="Run test suite (centos7, pip, openmpi, 3.9)"
+      - status-success="Run test suite (centos7, pip, mpich, 3.7)"
+      - status-success="Run test suite (centos7, pip, mpich, 3.8)"
+      - status-success="Run test suite (centos7, pip, mpich, 3.9)"
+      - status-success="Run test suite (rockylinux8, pip, openmpi, 3.7)"
+      - status-success="Run test suite (rockylinux8, pip, openmpi, 3.8)"
+      - status-success="Run test suite (rockylinux8, pip, openmpi, 3.9)"
+      - status-success="Run test suite (rockylinux8, pip, mpich, 3.7)"
+      - status-success="Run test suite (rockylinux8, pip, mpich, 3.8)"
+      - status-success="Run test suite (rockylinux8, pip, mpich, 3.9)"
+      - status-success="Run test suite (rockylinux8, conda, openmpi, 3.7)"
+      - status-success="Run test suite (rockylinux8, conda, mpich, 3.7)"
+      - status-success="Run test suite (ubuntu2004, pip, openmpi, 3.7)"
+      - status-success="Run test suite (ubuntu2004, pip, openmpi, 3.8)"
+      - status-success="Run test suite (ubuntu2004, pip, openmpi, 3.9)"
+      - status-success="Run test suite (ubuntu2004, pip, mpich, 3.7)"
+      - status-success="Run test suite (ubuntu2004, pip, mpich, 3.8)"
+      - status-success="Run test suite (ubuntu2004, pip, mpich, 3.9)"
```

Comment:
All these tests seem like overkill to me. Couldn't we just pick the minimum version and test against that in our CI? Or maybe just pick a low and a high version, like 3.7 and 3.10?

Reply:
I agree, it is a lot of tests to be run on one go, especially as we more and more to the build matrix. I think that ideally we would want to have a potentially large build matrix, but then actually build and test on a random subset of those on a somewhat regular basis as a means of collecting data for the dashboard. I am not sure where that leaves us for now though